SharedLibs/TileMapHelper/src/render/TilemapPixiHelper.ts

```diff
@@ -0,0 +1,274 @@
+   * @param layerIndex If `displayMode` is set to `index`, the layer index to be displayed.
+   */
+  static updatePixiTileMap(
+    pixiTileMap: any,
+    tileMap: EditableTileMap,
+    textureCache: TileTextureCache,
+    displayMode: "index" | "visible" | "all",
+    layerIndex: number
+  ): void {
+    if (!pixiTileMap) return;
+    pixiTileMap.clear();
+
+    for (const layer of tileMap.getLayers()) {
+      if (
+        (displayMode === "index" && layerIndex !== layer.id) ||
+        (displayMode === "visible" && !layer.isVisible())
+      ) {
+        return;
+      }
+
+      if (layer instanceof EditableObjectLayer) {
+        const objectLayer = layer as EditableObjectLayer;
+        for (const object of objectLayer.objects) {
+          const texture = textureCache.findTileTexture(
+            object.getTileId(),
+            object.isFlippedHorizontally(),
+            object.isFlippedVertically(),
+            object.isFlippedDiagonally()
+          );
+          if (texture) {
+            pixiTileMap.addFrame(
+              texture,
+              object.x,
+              object.y - objectLayer.tileMap.getTileHeight()
+            );
+          }
+        }
+      } else if (layer instanceof EditableTileMapLayer) {
+        const tileLayer = layer as EditableTileMapLayer;
+
+        for (let y = 0; y < tileLayer.tileMap.getDimensionY(); y++) {
+          for (let x = 0; x < tileLayer.tileMap.getDimensionX(); x++) {
+            const tileWidth = tileLayer.tileMap.getTileWidth();
+            const xPos = tileWidth * x;
+            const yPos = tileLayer.tileMap.getTileHeight() * y;
+
+            const tileId = tileLayer.get(x, y)!;
+
+            const tileTexture = textureCache.findTileTexture(
+              tileId,
+              tileLayer.isFlippedHorizontally(x, y),
+              tileLayer.isFlippedVertically(x, y),
+              tileLayer.isFlippedDiagonally(x, y)
+            );
+            if (tileTexture) {
+              const pixiTilemapFrame = pixiTileMap.addFrame(
+                tileTexture,
+                xPos,
+                yPos
+              );
+
+              const tileDefinition = tileLayer.tileMap.getTileDefinition(
+                tileId
+              );
+
+              // Animated tiles have a limitation:
+              // they are only able to use frames arranged horizontally one next
+              // to each other on the atlas.
+              if (tileDefinition && tileDefinition.getAnimationLength() > 0) {
+                pixiTilemapFrame.tileAnimX(
+                  tileWidth,
+                  tileDefinition.getAnimationLength()
+                );
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Re-renders the collision mask
+   *
+   * @param pixiGraphics
+   * @param tileMap
+   * @param typeFilter
+   * @param outlineSize
+   * @param outlineColor
+   * @param outlineOpacity
+   * @param fillColor
+   * @param fillOpacity
```

Comment:
Either enter a description or don't list these

Reply:
Will the documentation generation get the right method signature?

Reply:
Yes, that signature information can already be gotten through static analysis, and our current documentation generator relies on typescript to get the function signature.